src/test/java/de/zalando/aruha/nakadi/controller/EventStreamControllerTest.java

```diff
@@ -0,0 +1,280 @@
+import static javax.ws.rs.core.Response.Status.SERVICE_UNAVAILABLE;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup;
+import static org.zalando.problem.MoreStatus.UNPROCESSABLE_ENTITY;
+
+public class EventStreamControllerTest {
+
+    private static final String TEST_EVENT_TYPE = "test";
+
+    private NativeWebRequest requestMock;
+    private HttpServletResponse responseMock;
+    private TopicRepository topicRepositoryMock;
+    private EventStreamFactory eventStreamFactoryMock;
+
+    private ObjectMapper objectMapper;
+    private EventStreamController controller;
+    private JsonTestHelper jsonHelper;
+
+    @Before
+    public void setup() {
+        objectMapper = new NakadiConfig().jacksonObjectMapper();
+        jsonHelper = new JsonTestHelper(objectMapper);
+
+        topicRepositoryMock = mock(TopicRepository.class);
+        eventStreamFactoryMock = mock(EventStreamFactory.class);
+
+        controller = new EventStreamController(topicRepositoryMock, objectMapper,
+                eventStreamFactoryMock);
+
+        requestMock = mock(NativeWebRequest.class);
+        responseMock = mock(HttpServletResponse.class);
+    }
+
+    @Test
+    public void whenNoParamsThenDefaultsAreUsed() throws Exception {
+        final ArgumentCaptor<EventStreamConfig> configCaptor = ArgumentCaptor.forClass(EventStreamConfig.class);
+        final EventStream eventStreamMock = mock(EventStream.class);
+        when(eventStreamFactoryMock.createEventStream(any(), any(), configCaptor.capture()))
+                .thenReturn(eventStreamMock);
+
+        when(topicRepositoryMock.topicExists(eq(TEST_EVENT_TYPE))).thenReturn(true);
+        when(topicRepositoryMock.areCursorsValid(eq(TEST_EVENT_TYPE), any())).thenReturn(true);
+
+        final MockMvc mockMvc = standaloneSetup(controller)
+                .setMessageConverters(new StringHttpMessageConverter(),
+                        new MappingJackson2HttpMessageConverter(objectMapper))
+                .build();
+
+        mockMvc.perform(
+                get(String.format("/event-types/%s/events", TEST_EVENT_TYPE))
+                        .header("X-nakadi-cursors", "[{\"partition\":\"0\",\"offset\":\"0\"}]"))
+                .andExpect(status().isOk());
+
+        // we have to sleep here as mockMvc exits at the very beginning, before the body starts streaming
+        Thread.sleep(2000);
```

Comment:
Maybe we could use our retry strategy in order to assert this as fast as possible. 2000 timeout and 50 between retries. Wdyt?